From 4165d852d1abaeecb29be0ca3ba1d9a455b0794e Mon Sep 17 00:00:00 2001
From: Andrii Beregovenko <a.beregovenko@ti.com>
Date: Wed, 8 Aug 2012 18:23:55 +0300
Subject: [PATCH] WifiWatchdog: handle exception from ArpPeer

Sometimes IllegalArgumantException can happen in constructor of ArpPeer
class when IPv6 address is provided. This causes crash of systemserver.

Fixing it by adding appropriate exception handler

Change-Id: I4bb7bbca790745a1bf56ece96dd91b84b1f70cad
Signed-off-by: Andrii Beregovenko <a.beregovenko@ti.com>
Signed-off-by: Vishal Mahaveer <vishalm@ti.com>
---
 .../android/net/wifi/WifiWatchdogStateMachine.java |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/wifi/java/android/net/wifi/WifiWatchdogStateMachine.java b/wifi/java/android/net/wifi/WifiWatchdogStateMachine.java
index 1a42f93..c6d3eae 100644
--- a/wifi/java/android/net/wifi/WifiWatchdogStateMachine.java
+++ b/wifi/java/android/net/wifi/WifiWatchdogStateMachine.java
@@ -881,6 +881,9 @@ public class WifiWatchdogStateMachine extends StateMachine {
             //test to avoid any wifi connectivity issues
             loge("ARP test initiation failure: " + se);
             success = true;
+        } catch (IllegalArgumentException ae) {
+            log("ARP test initiation failure: " + ae);
+            success = true;
         }
 
         return success;
-- 
1.7.9

